File: src/test/java/mediathek/tool/table/MVTableTest.java
/*
 * Copyright (c) 2025 derreisende77.
 * This code was developed as part of the MediathekView project https://github.com/mediathekview/MediathekView
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package mediathek.tool.table;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MVTableTest {

    @Test
    void countNumberOfColumns() {
        var count = MVTable.countNumberOfColumns("a,b,c,d");
        assertEquals(4, count);

        count = MVTable.countNumberOfColumns("a,b,c,d,e");
        assertEquals(5, count);
    }
}

File: src/test/kotlin/mediathek/tool/table/MVTableTest.kt
package mediathek.tool.table

import mediathek.controller.DownloadColumn
import mediathek.daten.DatenDownload
import mediathek.tool.models.TModelDownload
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class MVTableTest {
    @Test
    fun countNumberOfColumns() {
        var count = PersistentColumnConfigurationTable.countNumberOfColumns("a,b,c,d")
        assertEquals(4, count)

        count = PersistentColumnConfigurationTable.countNumberOfColumns("a,b,c,d,e")
        assertEquals(5, count)
    }

    @Test
    fun downloadModelReturnsLiveValuesForColumnsLoadedWhileHidden() {
        val download = DatenDownload()
        download.aboName = "Daily Abo"
        download.sender = "ZDF"

        val row = Array<Any>(DownloadColumn.COUNT) { "" }
        row[DownloadColumn.REF.index] = download

        val model = TModelDownload()
        model.addRow(row)

        assertEquals(DownloadColumn.COUNT, model.columnCount)
        assertEquals("Abo", model.getColumnName(DownloadColumn.ABO.index))
        assertEquals("Größe [MB]", model.getColumnName(DownloadColumn.SIZE.index))
        assertEquals(String::class.java, model.getColumnClass(DownloadColumn.ABO.index))
        assertEquals(Boolean::class.javaObjectType, model.getColumnClass(DownloadColumn.DEFERRED.index))

        assertEquals("Daily Abo", model.getValueAt(0, DownloadColumn.ABO.index))
        assertEquals("ZDF", model.getValueAt(0, DownloadColumn.SENDER.index))
    }
}
